### PR TITLE
Genesys all queues together

### PR DESCRIPTION
**Pull request by Diego-H-S**
## Summary
In order to be more general when calling Genesys API, we have changed the method `genesys_generate_exports` and renamed it as `genesys_api_connection` introducing a new parameter `end_point`. Also, it has been added a `method` parameter to choose between wither connections, `POST` or `GET`.
It has been removed old methods never used in production but as a testing concept in the very early beginning of the actual project.
Furthermore, some parts of the calls have been modified, adding some time sleep step, to not overwhelm Genesys API, otherwise, we get fewer responses than calls made. 

## Importance
Required by the client to retrieve all queues from Genesys at once. The actual paradigm creates one call per queue.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: CHANGELOG.md
Comment (Rafalz13) [inline CHANGELOG.md:1]: Please check your branch version. You added comments above `0.4.13`.

File: viadot/sources/genesys.py [hunk 1]
```diff
@@ -152,33 +135,38 @@ def authorization_token(self, verbose: bool = False):
...
         return request_headers
 
-    def genesys_generate_exports(
-        self, post_data_list: List[str], end_point: str = "reporting/exports"
+    def genesys_api_connection(
+        self,
+        post_data_list: List[str],
+        end_point: str = "analytics/reporting/exports",
+        params: Dict[str, Any] = None,
+        method: str = "POST",
```
Comment (Rafalz13) [inline viadot/sources/genesys.py:143]: Add information on what options are possible for this argument. `POST` and `GET`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 ### Added
 - Added `anonymize_df` task function to `task_utils.py` to anonymize data in the dataframe in selected columns.
+- Added `GET` connection inside the method `genesys_api_connection`.
+
+### Changed
+- Changed the method to call Genesys API and its name, providing a new paremeter `end_point` to be more general.
+Name changed from `genesys_generate_exports` to `genesys_api_connection`.
```
Comment (Rafalz13) [inline CHANGELOG.md:14]: Please rephrase to be more clear, below is the example:
Genesys API call method and the name changed from `genesys_generate_exports` to `genesys_api_connection`. Introduced a new `end_point` parameter to make it more generic.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Changed Genesys source file, deleting methods for testing purposes but never used in production.
```
Comment (Rafalz13) [inline CHANGELOG.md:15]: Please correct. -> `Removed methods never used in production: 'methods..' ,.. `

**Review by Rafalz13 — COMMENTED** [review]
Please check the Changelog file and your branch version.

File: CHANGELOG.md (hunk 3 above)
Comment (Diego-H-S) [reply CHANGELOG.md:15]: the less the better, done!

File: CHANGELOG.md (hunk 2 above)
Comment (Diego-H-S) [reply CHANGELOG.md:14]: I'm not very good at redacting sentences, so I agree with your proposal.

**Review by Diego-H-S — COMMENTED** [review]
added all the comments :)

File: CHANGELOG.md
Comment (Diego-H-S) [reply CHANGELOG.md:1]: I need to rebase...

**Review by Diego-H-S — COMMENTED** [review]
all corrections added.

File: viadot/sources/genesys.py [hunk 4]
```diff
@@ -323,21 +315,17 @@ def download_all_reporting_exports(
...
-            if single_report[4].lower() == "queue_performance_detail_view":
-                file_name = (
-                    temp_ids_mapping.get(single_report[2]) + "_" + single_report[3]
-                ).upper()
+            date = self.start_date.replace("-", "")
+            if single_report[4].lower() in [
+                "queue_performance_detail_view",
+                "agent_status_detail_view",
+            ]:
+                file_name = self.view_type.upper() + f"_{next(self.count)}_" + f"{date}"
```
Comment (Rafalz13) [inline viadot/sources/genesys.py:323]: You can use also fstring in this way: `f"{self.view_type.upper()}_{next(self.count)}_{date}"`

**Review by Rafalz13 — COMMENTED** [review]
Looks good

File: viadot/sources/genesys.py (hunk 4 above)
Comment (Diego-H-S) [reply viadot/sources/genesys.py:323]: You are right, this is a legacy from the first version. Changed

**Pull request merged**